packages/flutter_tools/lib/src/android/gradle.dart

```diff
@@ -205,6 +205,7 @@ class AndroidGradleBuilder implements AndroidBuilder {
       isBuildingBundle: false,
       localGradleErrors: gradleErrors,
       configOnly: configOnly,
+      maxRetries: 1,
```

Comment:
Fun fact, since this defaults to null, we were never really retrying on these gradle errors.